File: manual/data/cpp/tests/test_autopep8_fix_file.cpp
#include <gtest/gtest.h>
#include "../src/function_autopep8_fix_file.cpp"

class TestClangFormatter : public ::testing::Test {
protected:
    ClangFormatter formatter;
};

TEST_F(TestClangFormatter, FormatCodeBasic) {
    std::string code = "def bad_function(x,y):print(x+y)\n";
    std::string expected = "def bad_function(x, y) : print(x + y)\n";
    EXPECT_EQ(formatter.format_code(code), expected);
}

TEST_F(TestClangFormatter, FormatCodeWithEmptyLines) {
    std::string code = "\ndef bad_function(x,y):print(x+y)\n\ndef another_function(a,b):print(a+b)\n";
    std::string expected = "\ndef bad_function(x, y)\n    : print(x + y)\n\n          def another_function(a, b)\n    : print(a + b)\n";
    EXPECT_EQ(formatter.format_code(code), expected);
}

TEST_F(TestClangFormatter, FormatCodeMixedIndentation) {
    std::string code = "\ndef bad_function(x, y):\n\tprint(x+y)\n";
    std::string expected = "\ndef bad_function(x, y) : print(x + y)\n";
    EXPECT_EQ(formatter.format_code(code), expected);
}

TEST_F(TestClangFormatter, FormatCodeNoFunction) {
    std::string code = "\nx = 10\ny = 20\nz = x + y\nprint(z)\n";
    std::string expected = "\nx = 10 y = 20 z = x + y print(z)\n";
    EXPECT_EQ(formatter.format_code(code), expected);
}

TEST_F(TestClangFormatter, FormatCodeSingleLine) {
    std::string code = "x=1;y=2;z=x+y;print(z)\n";
    std::string expected = "x = 1;\ny = 2;\nz = x + y;\nprint(z)\n";
    EXPECT_EQ(formatter.format_code(code), expected);
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: code/test_suites/python/python/test_autopep8_fix_file.py
import unittest
from function_autopep8_fix_file import AutoPep8Formatter

class TestAutoPep8(unittest.TestCase):

    def setUp(self):
        self.formatter = AutoPep8Formatter()

    def test_format_python_code_basic(self):
        code = "def bad_function(x,y):print(x+y)\n"
        expected = "def bad_function(x, y): print(x+y)\n"
        self.assertEqual(self.formatter.format_code(code), expected)

    def test_format_python_code_with_empty_lines(self):
        code = "\ndef bad_function(x,y):print(x+y)\n\ndef another_function(a,b):print(a+b)\n"
        expected = "\ndef bad_function(x, y): print(x+y)\n\n\ndef another_function(a, b): print(a+b)\n"
        self.assertEqual(self.formatter.format_code(code), expected)

    def test_format_python_code_mixed_indentation(self):
        code = "\ndef bad_function(x, y):\n\tprint(x+y)\n"
        expected = "\ndef bad_function(x, y):\n    print(x+y)\n"
        self.assertEqual(self.formatter.format_code(code), expected)

    def test_format_python_code_no_function(self):
        code = "\nx = 10\ny = 20\nz = x + y\nprint(z)\n"
        expected = "\nx = 10\ny = 20\nz = x + y\nprint(z)\n"
        self.assertEqual(self.formatter.format_code(code), expected)

    def test_format_python_code_single_line(self):
        code = "x=1;y=2;z=x+y;print(z)\n"
        expected = "x = 1\ny = 2\nz = x+y\nprint(z)\n"
        self.assertEqual(self.formatter.format_code(code), expected)

if __name__ == "__main__":
    unittest.main()
